proto/src/main/proto/Rss.proto

```diff
@@ -72,11 +73,27 @@ message GetShuffleDataRequest {
   int32 partitionNum = 5;
   int32 readBufferSize = 6;
   int32 segmentIndex = 7;
+  int64 offset = 8;
+  int32 length = 9;
 }
 
 message GetShuffleDataResponse {
   bytes data = 1;
-  repeated ShuffleDataBlockSegment blockSegments = 2;
+  StatusCode status = 2;
+  string retMsg = 3;
+}
+
+message GetShuffleIndexRequest {
+  string appId = 1;
+  int32 shuffleId = 2;
+  int32 partitionId = 3;
+  int32 partitionNumPerRange = 4;
+  int32 partitionNum = 5;
+  int64 length = 6;
```

Comment:
why we need length?

Reply:
This is the size of index file size and it would be used when require memory to read index data

Reply:
index length should be got from server side with actual file size

Reply:
As discussed offline, add a interface to get file length from storage layer is not necessary. Remove this length field add a indexFileSizeHint with default value is 2m (big enough for an index file) to prevent memory exhaustion caused by highly concurrent index file requests. @colinmjj @jerqi